controllers/replication-controller/dellcsireplicationgroup_controller.go

```diff
@@ -478,3 +511,319 @@ func (r *ReplicationGroupReconciler) SetupWithManager(mgr ctrl.Manager, limiter
 		}).
 		Complete(r)
 }
+
+// Give a replication group name and target, swapAllPVC reassigns the PVC from local volume to remote volume.
+// It also retains the original reclaimPolicy and operates within a single cluster.
+func (r *ReplicationGroupReconciler) swapAllPVC(ctx context.Context, c connection.RemoteClusterClient, rgName string, rgTarget string, log logr.Logger) error {
+	pvcs, err := c.ListPersistentVolumeClaim(ctx, client.MatchingLabels{controller.ReplicationGroup: rgName})
+	if err != nil {
+		return fmt.Errorf("failed to list PVCs: %w", err)
+	}
+
+	var wg sync.WaitGroup
+	errChan := make(chan error, len(pvcs.Items))
+
+	for _, pvc := range pvcs.Items {
+		wg.Add(1)
+		go func(pvc v1.PersistentVolumeClaim) {
+			defer wg.Done()
+			err := r.swapPVC(ctx, c, pvc.Name, pvc.Namespace, pvc.Annotations[controller.RemotePV], rgTarget, log)
+			if err != nil {
+				errChan <- fmt.Errorf("error swapping PVC %s/%s: %s", pvc.Namespace, pvc.Name, err)
+			}
+		}(pvc)
+	}
+
+	go func() {
+		wg.Wait()
+		close(errChan)
+	}()
+
+	var errs []error
+	for err := range errChan {
+		errs = append(errs, err)
+	}
+
+	if len(errs) > 0 {
+		return fmt.Errorf("errors occurred while swapping PVCs: %s", errs)
+	}
+
+	return nil
+}
+
+func (r *ReplicationGroupReconciler) swapPVC(ctx context.Context, client connection.RemoteClusterClient, pvcName, namespace, targetPV, rgTarget string, log logr.Logger) error {
+	// Read the PVC
+	pvc, err := client.GetPersistentVolumeClaim(ctx, namespace, pvcName)
+	if err != nil {
+		return fmt.Errorf("error getting pv %s: %s", pvcName, err)
+	}
+
+	// Save the Reclaim Policy for both PVs - return reclaim policy to makepvcreclaimpolicyretain
+	pv, err := client.GetPersistentVolume(ctx, pvc.Spec.VolumeName)
+	if err != nil {
+		log.V(common.InfoLevel).Info(fmt.Sprintf("Error retrieving local PV %s: %s", pvc.Spec.VolumeName, err.Error()))
```

Comment:
should these be `info` level logs? It seems they would better fit the `error` level.